Signup page

BakriIWNL opened the pull request.

BakriIWNL committed:
Merge pull request #2 from BakriIWNL/oboarding

update signup with onboarding codes

BakriIWNL committed:
Merge branch 'signup' of https://github.com/BakriIWNL/internship_implementation into signup

BakriIWNL committed:
Merge branch 'oboarding' into signup

BakriIWNL committed:
completed signup page as a UI, will continue log in functionality after i setup login page

YousefSalahKassem commented on lib/core/components/customtextfield.dart:
```diff
@@ -1,32 +1,76 @@
+        borderRadius: BorderRadius.circular(16.r),
+      ),
+      child: TextFormField(
+        obscureText: (context.read<SignupCubit>().passwordVisible || !password) ? false : true,
```
if we have sign in and forget password you will add condition on it ?
don't make any dependency

BakriIWNL replied:
done

YousefSalahKassem commented on lib/core/components/light_purple_button.dart:
```diff
@@ -0,0 +1,32 @@
+  final Size size;
+
+  const LightPurpleButton({super.key,required this.text, required this.onPressed,required this.size});
+
```
you can make custom button and send color if u have 10 colors you will make buttons for each color ?

BakriIWNL replied:
done

YousefSalahKassem commented on lib/core/routes/routes.dart:
```diff
@@ -44,7 +45,14 @@ class Routes {
-          builder: (context, state) => const SignUpScreen()),
+          builder: (context, state) => MultiBlocProvider(
+            providers: [
+              BlocProvider(
```
it's single bloc why you need to use multi bloc ?

BakriIWNL replied:
done

YousefSalahKassem commented on lib/core/utils/app_strings.dart:
```diff
@@ -14,5 +14,13 @@ class AppStrings {
   //signup Strings
   static const String signUp = "Sign Up";
-
+  static const String email = "Email";
```
can you implement localization ?

BakriIWNL replied:
inshallah

YousefSalahKassem commented on lib/features/signup/presentation/cubits/signup/signup_cubit.dart:
```diff
@@ -0,0 +1,41 @@
+  void toggleCheckBox() {
+    checkboxValue = !checkboxValue;
+    debugPrint('Checkbox value: $checkboxValue');
+    emit(state.copyWith(state: GenericStates.updateUI));
```
imagine u will click on toggle check box and also u found password field is toggled too.
because you emit the same state

YousefSalahKassem commented on lib/features/signup/presentation/screens/signup_screen.dart:
```diff
@@ -1,91 +1,196 @@
-              fontWeight: FontWeight.w600,
+      ),
+      body: SingleChildScrollView(
+        child: Form(
```
for advice separate your code to widgets as u can make private widget for form and private widgets for actions and add all this in signup screen

BakriIWNL replied:
done

YousefSalahKassem commented on lib/features/signup/presentation/widgets/checkbox.dart:
```diff
@@ -0,0 +1,73 @@
+import 'package:itcores_internship_project/features/signup/presentation/cubits/signup/signup_cubit.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class CheckBox extends StatelessWidget {
```
make custom component for checkbox only and don't make it depend on feature 
it's just a shareable component

YousefSalahKassem reviewed (changes requested):
GJ BAK BAK
Please review this comments !

BakriIWNL committed:
initial commit for login

BakriIWNL committed:
did the changes requested (localization will be delayed)

BakriIWNL committed:
UI partial completion

BakriIWNL committed:
login and forgot pages are done
todo: 2 pages left in this feature, enter new password to reset, and sending an email
animations going back and forth through the application

YousefSalahKassem commented on lib/features/login/presentation/screens/login_screen.dart:
```diff
@@ -1,10 +1,184 @@
+                      size: Size(343.w, 56.h)),
+                  12.verticalSpace,
+                  Text(AppStrings.orWith,
+                      style: TextStyle(
```
just separate form and action for 2 private stateless widget 
_Form
_Actions

BakriIWNL replied:
done

YousefSalahKassem commented on lib/core/components/customtextfield.dart:
```diff
@@ -31,7 +30,7 @@ class NewCustomTextField extends StatelessWidget {
       ),
       child: TextFormField(
-        obscureText: (context.read<SignupCubit>().passwordVisible || !password) ? false : true,
+        obscureText: (password) ? false : true,
```
just put password ?

BakriIWNL replied:
fair enough

BakriIWNL committed:
applied changes

BakriIWNL committed:
should be final changes that fixes login and signup

BakriIWNL committed:
covers all the pages and proper navigation throughout, and made all states equatable
will now localize in arabic
